src/coreclr/jit/lsra.cpp

```diff
@@ -1571,11 +1573,19 @@ bool LinearScan::isRegCandidate(LclVarDsc* varDsc)
 #endif // FEATURE_SIMD
 
         case TYP_STRUCT:
+        {
             // TODO-1stClassStructs: support vars with GC pointers. The issue is that such
             // vars will have `lvMustInit` set, because emitter has poor support for struct liveness,
             // but if the variable is tracked the prolog generator would expect it to be in liveIn set,
             // so an assert in `genFnProlog` will fire.
-            return compiler->compEnregStructLocals() && !varDsc->HasGCPtr();
+            bool isRegCandidate = compiler->compEnregStructLocals() && !varDsc->HasGCPtr();
+#ifdef TARGET_LOONGARCH64
+            // The LoongArch64's ABI which the float args within a struct maybe passed by integer register
+            // when no float register left but free integer register.
+            isRegCandidate &= !genIsValidFloatReg(varDsc->GetOtherArgReg());
```

Comment:
This is still not addressed. Should be `isRegCandidate &= genIsValidIntReg(varDsc->GetOtherArgReg());

Reply:
Oh, sorry.
Maybe my previous reponse lost.

The `varDsc->GetOtherArgReg()` may be REG_NA which is ok.
So `!genIsValidFloatReg(varDsc->GetOtherArgReg())` is not equal `genIsValidIntReg()` ?

And I kept your first advice.